nomad/structs/structs.go

```diff
@@ -4954,6 +4957,12 @@ type Namespace struct {
 	ModifyIndex uint64
 }
 
+// NamespaceCapabilities represents a set of capabilities allowed for this
+// namespace, to be checked at job submission time.
+type NamespaceCapabilities struct {
+	EnabledTaskDrivers []string
+}
```

Comment:
From our internal discussion of the RFC, I think we're going to want to have both a allowlist and a denylist. So can we add a `DisabledTaskDrivers` here as well?

Reply:
This feels like the place where Apache's confusing `order by allow,deny allow from all` logic fits.  I think it should be you allow first, and then prune out of that set.